### PR TITLE
#162949204 Configure backend to use a Postgres database

### PR DESCRIPTION
**Pull request by Allan690**
#### What does this PR do?
Configures the backend of the application to use Postgres as it's database
#### Description of Task to be completed?
- Integrate a Postgres database and make the migrations
- add .env file with local db configurations
- Change database settings in the settings.py file
#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Make cloned directory the active directory: `cd ah-the-unsullied`
3. Checkout to the current branch: `git checkout ch-configure-postgres-db-162949204`
4. Set up a virtual environment and activate it.
```
- virtualenv <name_of_env>
- source <name_of_env>/bin/activate
```
5. Create a  .env file by doing `touch .env` in the current directory(root) and enter the database connection information as shown below: 
```
export DATABASE_URL=postgres://USERNAME:PASSWORD@HOST:5432/NAME
```
6. Create the database on terminal: `psql ah_unsullied`
7.  Export the environment variables: `source .env`
8. Make migrations: `python manage.py makemigrations && python manage.py migrate`
9. Create a super user: `python manage.py createsuperuser`
10. Run the server: `python manage.py runserver`
11. Login with the credentials of the super user by navigating to: `localhost:8000/admins`
#### Any background context you want to provide?
PostgreSQL is Open Source, fully featured and optimized for production-ready environments as compared to SQLite that comes as default with Django
#### What are the relevant pivotal tracker stories?
[#162949204](https://www.pivotaltracker.com/story/show/162949204)

### CODE REVIEW

**Review by kwanj-k — CHANGES_REQUESTED** [review]
Good work Allan
But I have a few concerns
since am also setting up the database in production/staging, it would be wise if we use dj-database-url
[more info here](https://github.com/kennethreitz/dj-database-url) so we can both use the same settings file.
With dj-database-url you will just pass a database URL variable

**Comment by Allan690** [comment]
Thanks, @kwanj-k.  Let me work on making those changes. I think it will be easier that way

**Comment by MbugwaSami** [comment]
Good work Allan

**Comment by kelvinwm** [comment]
5. Create a .env file by doing touch .env and enter the database connection information as shown below: - Where do we create the.env file?

**Comment by Allan690** [comment]
> 1. Create a .env file by doing touch .env and enter the database connection information as shown below: - Where do we create the.env file?

I just updated the description to show that the file is created in the root directory. Please check

**Review by kwanj-k — DISMISSED** [review]

**Review by nesh-dev — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by kelvinwm — DISMISSED** [review]

**Review by MbugwaSami — DISMISSED** [review]

**Comment by sylviawanjiku** [comment]
Update your documentation of the PR.

**Comment by Allan690** [comment]
> Update your documentation of the PR.

Just did so, Please check

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class UserManager(BaseUserManager):
     """
     Django requires that custom users define their own Manager class. By
     inheriting from `BaseUserManager`, we get a lot of the same code used by
-    Django to create a `User` for free.
```
Comment (indungu) [inline authors/apps/authentication/models.py]: Remove from changed files.

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def validate(self, data):
 class UserSerializer(serializers.ModelSerializer):
     """Handles serialization and deserialization of User objects."""
 
-    # Passwords must be at least 8 characters, but no more than 128
```
Comment (indungu) [inline authors/apps/authentication/serializers.py]: Remove from changed files

File: authors/apps/authentication/views.py [hunk 3]
```diff
@@ -72,4 +72,3 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
-
```
Comment (indungu) [inline authors/apps/authentication/views.py]: Remove from changed files.

File: authors/urls.py [hunk 4]
```diff
@@ -20,6 +20,5 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('api/', include('authors.apps.authentication.urls', namespace='authentication')),
-
```
Comment (indungu) [inline authors/urls.py]: Remove from changed files.

**Review by indungu — CHANGES_REQUESTED** [review]
Please remove all this lint related file changes. The only files required for this PR are the `authors/settings.py` and the `requirements.txt` file

**Comment by Allan690** [comment]
Closing this PR due to unresolvable conflicts

**Pull request closed without merging**